### PR TITLE
allow non-ascii characters in printf format strings

### PR DESCRIPTION
**Pull request by KlausC**
Fix https://github.com/JuliaLang/julia/issues/23880

### CODE REVIEW

File: base/printf.jl [hunk 1]
```diff
@@ -36,23 +36,24 @@ end
 ### printf format string parsing ###
 
 function parse(s::AbstractString)
-    # parse format string in to stings and format tuples
+    # parse format string into strings and format tuples
     list = []
     i = j = start(s)
+    j1 = 0 # invariant: j1 == prevind(s, j-1)
```
Comment (bkamins) [inline base/printf.jl]: @KlausC minor issue; if I understand it correctly the comment should read:
```
# invariant: j1 == prevind(s, j)
```

**Comment by yuyichao** [comment]
Please use a more descriptive commit message and PR title. (title and top post corrected)

File: base/printf.jl (hunk 1 above)
Comment (KlausC) [reply base/printf.jl]: you are right!

**Review by nalimilan — APPROVED** [review]
@bkamins Good to merge?

**Review by bkamins — APPROVED** [review]

**Comment by bkamins** [comment]
@nalimilan yes

**Pull request merged**